c10/util/DeadlockDetection.cpp

```diff
@@ -0,0 +1,21 @@
+#include <c10/util/DeadlockDetection.h>
+
+namespace c10 {
+namespace impl {
+
+namespace {
+PythonGILHooks* python_gil_hooks = nullptr;
+}
+
+bool check_python_gil() {
```

Comment:
nit: "check" does not give any information if we expect it to be held or not.
Maybe `get_python_gil_state()`?

Reply:
I'm drawing analogy with the analogous CPython API `PyGILState_Check`. BTW, "get" is bad because it aliases with similar sounding `PyGILState_GetThisThreadState` and `PyThreadState_Get`

Reply:
Also the name here doesn't really matter, it's not public API

Reply:
Similarity with `PyGILState_Check` sounds good :)